src/components/Header/Header.tsx

```diff
@@ -0,0 +1,121 @@
+import { useState } from 'react';
+import cn from 'classnames';
+
+import { addTodo, updateTodo } from '../../api/todos';
+
+import { useTodosContext } from '../../context/TodosContext';
+
+import { temporaryTodo } from '../../utils/tempTodo';
+
+import { AddTodoForm } from './AddTodoForm';
+
+import { Todo } from '../../types/Todo';
+
+export const Header = () => {
+  const [title, setTitle] = useState('');
+  const { todos, setTodos, setErrorMessage, setTempTodo, setLoadingIds } =
+    useTodosContext();
+
+  const isToggleButtonActive = todos.every(t => t.completed);
+  const showToggleAll = todos.length > 0;
+
+  function createNewTodo(currentTitle: string) {
+    const newTodo: Todo = {
+      ...temporaryTodo,
+      title: currentTitle.trim(),
+    };
+
+    return newTodo;
+  }
+
+  function addNewTodo(newTodo: Todo) {
+    addTodo(newTodo)
+      .then(todoFromServer => {
+        setTodos(prev => [...prev, todoFromServer]);
+        setTitle('');
+      })
+      .catch(() => {
+        setErrorMessage('Unable to add a todo');
+        setTempTodo(null);
+      })
+      .finally(() => {
+        setTempTodo(null);
+        setLoadingIds([]);
+      });
+  }
+
+  function updatingTodo(todoForUpdate: Todo, updatedTodos: Todo[]) {
```

Comment:
update todo logic can also be in the context (naming needs to be fixed)

```suggestion
  function updateTodo(todoForUpdate: Todo, updatedTodos: Todo[]) {
```
```suggestion
  function updatingTodo(todoForUpdate: Todo, updatedTodos: Todo[]) {
```